src/client-agent/notify.c

```diff
@@ -39,42 +39,32 @@ char *getsharedfiles()
     return (ret);
 }
 
-#ifndef WIN32
+
 char *get_agent_ip()
 {
     char agent_ip[IPSIZE + 1] = { '\0' };
-#if defined (__linux__) || defined (__MACH__) || defined (sun) || defined(FreeBSD) || defined(OpenBSD)
-    int sock;
-    int i;
-    static const char * REQUEST = "host_ip";
-
-    for (i = SOCK_ATTEMPTS; i > 0; --i) {
-        if (sock = control_check_connection(), sock >= 0) {
-            if (OS_SendUnix(sock, REQUEST, strlen(REQUEST)) < 0) {
-                mdebug1("Error sending msg to control socket (%d) %s", errno, strerror(errno));
-            }
-            else{
-                if (OS_RecvUnix(sock, IPSIZE, agent_ip) <= 0) {
-                    mdebug1("Error receiving msg from control socket (%d) %s", errno, strerror(errno));
-                    agent_ip[0] = '\0';
-                }
-            }
-
-            close(sock);
-            break;
-        } else {
-            mdebug2("Control module not yet available. Remaining attempts: %d", i - 1);
-            sleep(1);
+    struct sockaddr_storage sas;
+    socklen_t len = sizeof(sas);
+    const int err = getsockname(agt->sock, (struct sockaddr *)&sas, &len);
+
+    if (!err) {
+        switch (sas.ss_family) {
+            case AF_INET:
+                get_ipv4_string(((struct sockaddr_in *)&sas)->sin_addr, agent_ip, IPSIZE);
+                break;
+            case AF_INET6:
+                get_ipv6_string(((struct sockaddr_in6 *)&sas)->sin6_addr, agent_ip, IPSIZE);
+                break;
+            default:
+                mdebug2("Unknown address family: %d", sas.ss_family);
+                break;
         }
+    } else {
+        mdebug2("getsockname() failed: %s", strerror(errno));
```

Comment:
As I commented before, using the same code for both Linux and Windows agents without any defines or comments, is error prone. It also can make the reviewer to ignore some details.

Here, for example, when `getsockname()` fails the variable **errno** is set in Linux (https://man7.org/linux/man-pages/man2/getsockname.2.html) but in Windows we should call `WSAGetLastError()` instead (https://learn.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-getsockname)